invenio_app_ils/ui/src/invenio_app_ils/forms/components/SelectorField.js

```diff
@@ -0,0 +1,161 @@
+import { ESSelector } from '../../common/components/ESSelector';
+import isEmpty from 'lodash/isEmpty';
+
+export class SelectorField extends Component {
+  constructor(props) {
+    super(props);
+
+    const {
+      fieldPath,
+      errorPath,
+      emptyHeader,
+      emptyDescription,
+      label,
+      required,
+      multiple,
+      serializer,
+      renderGroup,
+      renderSelection,
+      ...selectorProps
+    } = props;
+    this.fieldPath = fieldPath;
+    this.errorPath = errorPath;
+    this.emptyHeader = emptyHeader;
+    this.emptyDescription = emptyDescription;
+    this.label = label;
+    this.required = required;
+    this.multiple = multiple;
+    this.renderGroup = renderGroup || this.defaultRenderGroup;
+    this.renderSelection = renderSelection || this.defaultRenderSelection;
+    this.serializer = serializer;
+    this.selectorProps = selectorProps;
+
+    this.state = {
+      value: null,
+    };
+  }
+
+  renderEmpty = () => (
+    <Card header={this.emptyHeader} description={this.emptyDescription} />
+  );
+
+  defaultRenderSelection = (selection, removeSelection) => {
+    return (
+      <Card fluid={!this.multiple} key={selection.id}>
+        <Card.Content>
+          <Card.Header as="a" onClick={() => removeSelection(selection)}>
+            {selection.title}
+            <Icon color="red" name="delete" />
+          </Card.Header>
+          <Card.Meta>{selection.extra}</Card.Meta>
+          <Card.Description>{selection.description}</Card.Description>
+        </Card.Content>
+      </Card>
+    );
+  };
+
+  defaultRenderGroup = (selections, renderSelection, removeSelection) => {
+    if (isEmpty(selections)) {
+      return this.renderEmpty();
+    }
+
+    return (
+      <Card.Group>
+        {selections.map(selection =>
+          renderSelection(selection, removeSelection)
+        )}
+      </Card.Group>
+    );
+  };
+
+  onSelectionsUpdate = (selections, setFieldValue) => {
+    this.setState({ value: selections });
+    if (this.multiple) {
+    } else {
+      setFieldValue(
+        this.fieldPath,
+        selections.length > 0 ? selections[0].metadata : {}
+      );
+    }
+  };
+
+  hasFieldError(errors, name, value) {
+    const error = errors[name];
+    return !isEmpty(error);
+  }
+
+  renderFormField = ({ form: { errors, setFieldValue, values } }) => {
+    const selections = [];
+    const value = values[this.fieldPath];
+    if (this.multiple) {
+      for (const record of value) {
+        if (!isEmpty(record)) {
+          selections.push(this.serializer({ metadata: record }));
+        }
+      }
+    } else {
+      if (!isEmpty(value)) {
+        selections.push(this.serializer({ metadata: value }));
+      }
+    }
+    const hasFieldError = this.hasFieldError(errors, this.errorPath, value);
+    let error = null;
+    if (errors[this.errorPath]) {
+      error = errors[this.errorPath];
+    } else {
+      if (this.required) {
```

Comment:
If for any reason we end up with errors not having the format `[errorPath]: [message]` and the field is not required we will have a misleading message. If the field is required the error will come in the correct format from the loaders, no?

Reply:
Probably I haven't understood why you need this extra check? :P

Reply:
Yes, this was a left-over from what we discussed yesterday about the live validation for required fields. :)